### PR TITLE
feat(perf): add base for scenarios

### PR DESCRIPTION
**Pull request by majorgreys**
## Description

This PR takes the approach used in #2646 as the base for all scenarios. You can also now specify a git source or to install from pypi.

## Checklist
- [ ] Added to the correct milestone.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

**Review by Kyle-Verhoog — APPROVED** [review]

File: benchmarks/base/benchmark [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+RESULTS_V1=${ARTIFACTS}/${DDTRACE_V1}.json
+RESULTS_V2=${ARTIFACTS}/${DDTRACE_V2}.json
+
+# append venvs with ddtrace to sys.path
+
+PYTHONPATH=${VENV_DDTRACE_V1}/lib/python3.9/site-packages \
+    python scenario.py \
+    --copy-env \
+    --fast \
+    -o ${RESULTS_V1}
```
Comment (brettlangdon) [inline benchmarks/base/benchmark:19]: should we also do `${VENV_DDTRACE_V1/bin/python` here?

File: benchmarks/README.rst [hunk 2]
```diff
@@ -4,12 +4,13 @@ Performance Testing
 New scenario
 ------------
 
-Copy ``sample`` directory and edit the ``meta.yaml`` file to specify the executable. Add any additional requirements to the ``requirements.txt``.
+Create a new directory for a scenario which includes a ``scenario.py`` script to be run. Add any additional dependencies in a ``requirements_scenario.py``.
```
Comment (brettlangdon) [inline benchmarks/README.rst:7]:
```suggestion
Create a new directory for a scenario which includes a ``scenario.py`` script to be run. Add any additional dependencies in a ``requirements_scenario.txt``.
```

?

**Review by brettlangdon — COMMENTED** [review]
some small nits, but otherwise lgtm

File: benchmarks/base/benchmark (hunk 1 above)
Comment (majorgreys) [reply benchmarks/base/benchmark:19]: This happens to be just symlinks to the global/system python version.

Comment (brettlangdon) [reply benchmarks/base/benchmark:19]: oh, easy enough then!

**Review by Yun-Kim — APPROVED** [review]

**Pull request merged**